pydatastructs/miscellaneous_data_structures/queue.py

```diff
@@ -88,20 +89,40 @@ def __new__(cls, items=None, dtype=NoneType):
         obj.items, obj.front = items, -1
         if items.size == 0:
             obj.front = -1
+            obj.rear = -1
         else:
             obj.front = 0
+            obj.rear = items._num - 1
+        obj.deque = double_ended
```

Comment:
Why different names? `obj.double_ended` should have worked.

Reply:
Sure a thing